contracts/src/utils/CallUtils.sol

```diff
@@ -0,0 +1,60 @@
+// (c) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+pragma solidity 0.8.18;
+
+library CallUtils {
+    /**
+     * @dev calls target address with exactly gasAmount gas and data as calldata
+     * or reverts if at least gasAmount gas is not available.
+     */
+    function _callWithExactGas(
```

Comment:
Obviously not a priority right now, but we can consider making the equivalent [`tryExecuteMessage`](https://github.com/ava-labs/teleporter/blob/main/contracts/src/Teleporter/TeleporterMessenger.sol#L789) public in a future release and use that here.